src/ngraph/runtime/cpu/pass/cpu_fusion.cpp

```diff
@@ -809,46 +809,62 @@ void ngraph::runtime::cpu::pass::CPUFusion::construct_batch_norm_relu_global_sta
 
-            auto pattern_map = m.get_pattern_map();
-            auto m_bn = std::dynamic_pointer_cast<op::BatchNorm>(
-                m.get_match_root()->get_inputs().at(0).get_output().get_node());
+        auto pattern_map = m.get_pattern_map();
 
-            // as of now, only MKLDNN supports this fusion
-            // and it requires input data's rank to be equal to 4
-            if (pattern_map[input]->get_shape().size() != 4)
-            {
-                NGRAPH_DEBUG << " Input data's rank isn't equal to 4. Shape = "
-                             << pattern_map[input]->get_shape().size();
-                return false;
-            }
+        // as of now, only MKLDNN supports this fusion
+        // and it requires input data's rank to be equal to 4
+        if (pattern_map[input]->get_shape().size() != 4)
+        {
+            NGRAPH_DEBUG << " Input data's rank isn't equal to 4. Shape = "
+                         << pattern_map[input]->get_shape().size();
+            return false;
+        }
 
-            if (m_bn->get_users().size() > 1)
-            {
-                NGRAPH_DEBUG << "Relu isn't the only user of BatchNorm's output";
-                return false;
-            }
+        auto bn_match = m.get_match_root()->get_inputs().at(0).get_output().get_node();
+        if (bn_match->get_users().size() > 1)
+        {
+            NGRAPH_DEBUG << "Relu isn't the only user of BatchNorm's output";
+            return false;
+        }
 
-            auto bn_relu = std::make_shared<op::BatchNormRelu>(m_bn->get_eps_value(),
-                                                               pattern_map[gamma],
-                                                               pattern_map[beta],
-                                                               pattern_map[input],
-                                                               pattern_map[mean],
-                                                               pattern_map[var],
-                                                               m_bn->get_training_flag());
+        std::shared_ptr<Node> bn_relu;
+        if (auto bn_inference = std::dynamic_pointer_cast<op::BatchNormInference>(bn_match))
+        {
+            bn_relu = std::make_shared<op::BatchNormInferenceRelu>(bn_inference->get_eps_value(),
+                                                                   pattern_map[gamma],
+                                                                   pattern_map[beta],
+                                                                   pattern_map[input],
+                                                                   pattern_map[mean],
+                                                                   pattern_map[var]);
+        }
+        else if (auto bn_training = std::dynamic_pointer_cast<op::BatchNormTraining>(bn_match))
+        {
+            bn_relu = std::make_shared<op::BatchNormTrainingRelu>(bn_training->get_eps_value(),
```

Comment:
this is little confusing, why are we replacing with `batchnorm relu` fused Op with global stats. It efficient  for bn training we compute the `global stats` within mkldnn kernel.